### ISSUE TITLE
Reproducing results from Table 1

### ISSUE THREAD

**Issue by neerajgangwar** [scale-lab/MTLoRA#6]
Hi @ahmed-agiza,

Based on Issues [#4](https://github.com/scale-lab/MTLoRA/issues/4) and [#5](https://github.com/scale-lab/MTLoRA/issues/5), I ran the code with the `mtlora_tiny_448_r64_scale4_pertask` config. I got the following results:
```bash
semseg: 68.38
humans: 59.58
sal: 63.85
normals: 16.76
delta_m: 1.77%
```
The validation was run every five epoch. The above-mentioned results are based on the best $\Delta m$ across all validation scores. The saliency performance seems to be the most off compared to the numbers in the paper, followed by normals. Any suggestions on what may be causing the discrepancy?

The task weights are the following:
```python
{
    'depth': 1.0,
    'semseg': 1.0,
    'human_parts': 2.0,
    'sal': 5.0,
    'edge': 50.0,
    'normals': 10.0,
}
```
Could you also check if you used the same weights for your runs? Also, could you please share a trained checkpoint when you get a chance, as discussed previously in [#4](https://github.com/scale-lab/MTLoRA/issues/4).

Thanks!

**Comment by ahmed-agiza** [comment]
Hi @neerajgangwar, it is a bit hard to debug an external run, but I can confirm that this is the same configuration yes; assuming you are running everything right, then the only reason I can think of is if we introduced some bug during the cleanup for releasing the public code (this the is only one we didn't get the chance to test thoroughly), so I can try to re-train using the public code after the conference to investigate (since my GPUs are busy for a while), even the smaller version (r=32) was never < 2. 
The exact command was `main.py --cfg configs/mtlora/tiny_448/mtlora_tiny_448_r64_scale4_pertask.yaml --pascal <dataset pat> --tasks semseg,normals,sal,human_parts --batch-size 32 --ckpt-freq=20 --epoch=300 --eval-freq 25 --resume-backbone <path>/swin_tiny_patch4_window7_224.pth`
And those are the results on wandb:
![image](https://github.com/scale-lab/MTLoRA/assets/6373629/6daa748b-2d5c-4b10-b506-2416089009c6)

I have the log file and the wandb run; for the checkpoint it is just in my co-author's home directory, and she is on a break till the conference, so hopefully, I can get it afterward (feel free to re-ping me!).

**Comment by neerajgangwar** [comment]
@ahmed-agiza Thanks for sharing the details. It would help identify the issue if you run the publicly released code and compare the results.  I will keep checking on my end. I am keeping this issue open for now. Thanks again!

**Comment by neerajgangwar** [comment]
@ahmed-agiza I was able to almost get the numbers reported in the paper.

| Task | Single Task Score | Mutli Task Score | Gain (`delta_m`) |
|------|------------------|------------------|------------------|
SemSeg (mIoU) |	67.21 |	68.32 |	1.65% |
Humans (mIoU) |	61.93 |	59.51 |	-3.91% |
Saliency (mIoU) |	62.35 |	65.49 |	5.04% |
Normals (rmse) |	17.97 |	16.75 |	6.79% |
Average | | |			2.39% |

I learned that a batch size of 16 with gradient accumulation for two steps does not result in a similar performance as a batch size of 32. The earlier results that I posted were with the former. I think the difference now may be just because of the seed values or other changes due to setup etc. What do you think?

**Comment by ahmed-agiza** [comment]
Hi @neerajgangwar, yeah, those numbers are much closer; I will still try to retrain when I get the chance to validate and post the weights to the repository, too.
I didn't try gradient accumulation in this project, but I do remember having a very similar problem before in a different project that was using Swin, too, but I didn't look into it back then.
Thanks for taking the time to investigate, and I will try to ping you when I post the weights.

**Comment by neerajgangwar** [comment]
Sure, thanks for the clarifications and help!
